cosmo/tests/test_serializer.py

```diff
@@ -81,10 +85,10 @@ def test_l2vpn_errors(capsys):
         'terminations': [
             { 'assigned_object': None },
             { 'assigned_object': { '__typename': None }}]})
-    serialize(unsupported_type_terminations)
-    assert "Found unsupported L2VPN termination in" in capsys.readouterr().err
+    with pytest.warns(UserWarning, match="Found unsupported L2VPN termination in"):
+        serialize(unsupported_type_terminations)
 
-    vpws_non_interface_term = template
+    vpws_non_interface_term = copy.deepcopy(template)
```

Comment:
why do we need a deepcopy here now?

Reply:
since template was a pointer, it was modified each time, appending more and more. this wasn't detected bc we were searching for the error string. when they piled up, the regexp didn't detected it. so there was a bug in the test :P